### PR TITLE
🧠 Logic: ⛓️ Implement `read_string/3` predicate

### PR DESCRIPTION
**Pull request by bdeneux**
Implement #380

### CODE REVIEW

**Review by amimart — CHANGES_REQUESTED** [review]
The implementation looks good to me :)

I just pointed out some grammatical mistakes in error messages, I've suggested some correction rephrasing messages, I'll let it to your appreciation :)

File: x/logic/predicate/string_test.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+				input:       "Hello World!",
+				program:     "read_input(String, Len) :- current_input(Stream), read_string(foo, Len, String).",
+				query:       `read_input(String, Len).`,
+				wantError:   fmt.Errorf("read_string/3: invalid domain for given stream"),
+				wantSuccess: false,
+			},
+			{
+				input:       "Hello World!",
+				query:       `read_string(Stream, Len, data).`,
+				wantError:   fmt.Errorf("read_string/3: stream could not be a variable"),
```
Comment (bdeneux) [inline x/logic/predicate/string_test.go:133]:
```suggestion
				wantError:   fmt.Errorf("read_string/3: stream cannot be a variable"),
```

**Review by amimart — APPROVED** [review]
🚀

**Review by ccamel — APPROVED** [review]
Great! Thanks :+1:

**Pull request merged**